changelog/11282.bugfix.rst

```diff
@@ -0,0 +1,3 @@
+Fixed returning "None" as the default value for a config option when "None" is explicitly
```

Comment:
While this is a bug fix, it is also a `breaking` change, so we should announce this accordingly.

Rename the file to `11282.breaking.rst` and use something like this for the contents:

```rst
Sanitized the default value for configuration options.

Previously if a default was not supplied during :meth:`parser.addini <pytest.Parser.addini>`, it would always return an *empty list* in case this option was not defined, which is clearly incorrect. Also ``None`` was not honored as ``default`` even if ``default=None`` was used explicitly while defining the option.

Now the behavior is as follows:

* If a default value is NOT set but the type for the option is provided, then a type specific default will be returned. For example ``type=bool`` will return ``False``, ``type=str`` will return ``""``, etc.
* If a default value of ``None`` is supplied, then ``None`` will be returned, regardless of the ``type``. This is controversial, as it can be argued that it should raise an error, but seems reasonable enough to avoid breaking code needlessly.
* If neither ``default`` or ``type`` are provided, assume ``type=str`` and return ``""`` as default (this is the previous behavior).

The team decided to not introduce a deprecation period for this change, as doing so would be complicated both in terms of communicating this to the community as well as implementing it, and also because the team believes this change should not break existing plugins except in rare cases.
```